concent_api/core/views.py

```diff
@@ -35,6 +36,8 @@
 @transaction.non_atomic_requests(using='control')
 def send(_request: HttpRequest, client_message: Message, client_public_key: bytes) -> Union[Message, HttpResponse]:
     assert isinstance(client_public_key, bytes) or client_public_key is None
+    if not are_all_stored_messages_compatible_with_protocol_version(client_message, client_public_key):
```

Comment:
1. This should be done in decorator too.
2. This should be done in `receive` too (`ClientAuthorization` might change and not be compatible too).
3. The function name is `are_all_stored_messages_compatible_with_protocol_version` but it check client message. The name should be consistent or more general.

Reply:
Ad.1 It is done in `check_protocol_versions_and_update_subtasks_from_incoming_message_if_timed_out` to reduce number of database queries. BTW it is one of many validations, why you think that this one should be in decorator? Validation that protocol version in header is compatible to concent is in decorator `handle_errors_and_responses`. This is something different
Ad. 2 It is in `handle_errors_and_responses` decorator
Ad. 3 This function has been removed
Ad. 4 It should be checked. E.g. client used protocol_version 2.15.0 and send message (stored_messages have version 2.15.0). Then he updated to 2.16.0 and send next message concerning the same subtask. This is incorrect and is checked.

Reply:
It shouldn't be done in `handle_errors_and_responses` - the decorator name points out what is used for. This should be a separate decorator, otherwise the will be some shared code between existing decorators. And making two versions of it, one for core and one for gatekeeper, can avoid making any views changes.

Reply:
done